static/src/javascripts/projects/common/modules/video/events.js

```diff
@@ -109,36 +110,50 @@ define([
         bindCustomMediaEvents(eventsMap, player, mediaId, mediaType, true);
     }
 
+    function getGoogleAnalyticsEventAction(mediaEvent) {
+        var action = mediaEvent.mediaType + ' ';
+        if (mediaEvent.isPreroll) {
+            action += 'preroll';
+        } else {
+            action += 'content';
+        }
+        return action;
+    }
+
+    function buildGoogleAnalyticsEvent(mediaEvent, metrics, canonicalUrl) {
+        var category = 'Media';
+        var playerName = 'guardian-videojs';
+        var action = getGoogleAnalyticsEventAction(mediaEvent);
+        var fieldsObject = {
+            eventCategory: category,
+            eventAction: action,
+            eventLabel: canonicalUrl,
+            dimension19: mediaEvent.mediaId,
+            dimension20: playerName
+        };
+        // Increment the appropriate metric based on the event type
+        var metricId = metrics[mediaEvent.eventType];
+        if (metricId) {
+            fieldsObject[metricId] = 1;
+        }
+        return fieldsObject;
+    }
 
-    function bindGoogleAnalyticsEvents(player) {
-        var allEvents = [
-            'ready',
-            'play',
-            'end',
-            'skip',
-            'watched25',
-            'watched50',
-            'watched75'
-        ];
-        var category = 'media';
-
-        // see ga() api here:
-        // https://developers.google.com/analytics/devguides/collection/analyticsjs/events#implementation
-        // ga('send', 'event', [eventCategory], [eventAction], [eventLabel], [eventValue], [fieldsObject]);
-        allEvents.map(function(eventName) {
+    function bindGoogleAnalyticsEvents(player, canonicalUrl) {
+        var events = {
+            'play': 'metric1',
```

Comment:
is it worth having some sort of shared config to map our event names to metricX so that we don't accidentally send media events to click events for example?